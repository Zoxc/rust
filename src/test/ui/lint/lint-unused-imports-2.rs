#![deny(unused_imports)]
#![allow(dead_code)]

// Be sure that if we just bring some methods into scope that they're also
// counted as being used.
use test::B;
// But only when actually used: do not get confused by the method with the same name.
use test::B2; //~ ERROR unused import: `test::B2`

mod test {
    pub trait B {
        fn b(&self) {}
    }
    pub trait B2 {
        fn b(&self) {}
    }
    pub struct C;
    impl B for C {}
}

fn main() {
    test::C.b();
}
