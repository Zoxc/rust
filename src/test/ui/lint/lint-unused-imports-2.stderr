error: unused import: `test::B2`
  --> $DIR/lint-unused-imports-2.rs:8:5
   |
LL | use test::B2;
   |     ^^^^^^^^
   |
note: lint level defined here
  --> $DIR/lint-unused-imports-2.rs:1:9
   |
LL | #![deny(unused_imports)]
   |         ^^^^^^^^^^^^^^

error: aborting due to previous error

